ocaml-lsp-server/src/state.ml

```diff
@@ -2,7 +2,7 @@ open Import
 
 type init =
   | Uninitialized
-  | Initialized of ClientCapabilities.t
+  | Initialized of InitializeParams.t
```

Comment:
Previously, we were trying to find the `_build` folder by walking up the tree based on the `cwd`. This caused an issue when I implemented the snapshot test because it was taking `cmt` files under the project's `_build` folder instead of the test folder (`__tests__/workspace_symbol`).

To overcome this issue, I had to [specify](https://github.com/ocaml/ocaml-lsp/pull/398/files#diff-c84ef09154bddc0a37a75301087282d6a921acfe31607b44cdde69b4ea38198bR13) the `rootUri` during the `initialize` request. Since on the server-side we didn't store this value, I had no way to use it. Hence why I now store the whole `InitializeParams.t` instead of just the `capabilities`.

Even without taking the tests into account, I think that it makes sense to keep the full `initializeParams`. Also because `roorUri` is deprecated and we'll need at some point to also use the `workspaceFolders` (introduced in 3.6.0)

Reply:
This sounds fine to me.